### ISSUE TITLE
Migrator for `xorg-xextproto` & `xorg-xproto` to `xorg-xorgproto`

### ISSUE THREAD

**Issue by jakirkham** [regro/cf-scripts#3360]
The packages `xorg-xextproto` & `xorg-xproto` have been deprecated

Users are recommended to migrate to `xorg-xorgproto`

The linter now warns about this situation: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6688

Would it make sense to put together a migrator of some kind to manage this transition with the bot?

**Comment by hmaarrfk** [comment]
There are a few other proto packages that can be caught by the migrator.
https://github.com/search?q=org%3Aconda-forge+xorg-+proto+language%3AYAML&type=code&l=YAML

once the core logic is in, then a cursory search through the first few pages might expand the ability to catch even more.

**Comment by hmaarrfk** [comment]
* `xorg-xineramaproto`
* ` xorg-xproto`
* `xorg-xf86vidmodeproto`
* `xorg-xextproto`
* `xorg-glproto`
* `xorg-inputproto`
* `xorg-kbproto`

**Comment by jakirkham** [comment]
Should we add linter hints for those as well?

**Comment by hmaarrfk** [comment]
i mean we could, but its somewhat "low importance". Don't we already have a migrator for package renaming?

we could migrate all the proto packages that we find in: https://conda-forge.org/packages/

**Comment by jakirkham** [comment]
Am not aware of an existing migrator. Hence the issue raised here to add one

At least currently see issues with clobbering when trying to install some packages from conda-forge. That would be nice to fix

**Comment by hmaarrfk** [comment]
yeah i misremembered sorry. it was the whole piggyback minimigrator stuff and complications...

**Comment by jakirkham** [comment]
So in general are we just want to migrate `xorg-*proto` to `xorg-xorgproto`?

**Comment by hmaarrfk** [comment]
correct.

**Comment by jakirkham** [comment]
Thanks Mark! 🙏

Updated the issue to reflect this

**Comment by jakirkham** [comment]
> Should we add linter hints for those as well?

Extended the hints to these packages and anything else referencing `xorg-*proto` except `xorg-xorgproto`

xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6816

**Comment by hmaarrfk** [comment]
see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6822 for how to implement such a migrator.

**Comment by jakirkham** [comment]
Think we would also need bot changes

For example: https://github.com/regro/cf-scripts/pull/3375

**Comment by hmaarrfk** [comment]
yes i shared the "link" above as a thread for somebody to pull on in case they were so inclined to implement things!

**Comment by beckermr** [comment]
Those migrators get applied to only some migrations and not others. We can have a single class, but we need to instantiate separate objects for each one with specific migrations in mind.

**Comment by mgorny** [comment]
Yeah, I see that now. Sorry, was still reading through the code. So yeah, I feel like a shared base class with the common logic should streamline things a bit.

**Comment by beckermr** [comment]
It is likely we only need a single class and not more than one. The Replacement migrator is customized on instantiating as opposed to making new child classes.

**Comment by mgorny** [comment]
Oh yes, looks like `Replacement` is pretty much what we need for "simple" cases. Though in case of `xorg-*proto`, we would probably want to deduplicate somehow.
